public/components/datasources/components/manage/associated_objects/modules/associated_objects_table.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+import {
+  EuiInMemoryTable,
+  EuiLink,
+  SearchFilterConfig,
+  EuiTableFieldDataColumnType,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { AssociatedObject } from '../../../../../../../common/types/data_connections';
+import {
+  getRenderAccelerationDetailsFlyout,
+  getRenderAssociatedObjectsDetailsFlyout,
+} from '../../../../../../plugin';
+import {
+  ASSC_OBJ_TABLE_ACC_COLUMN_NAME,
+  ASSC_OBJ_TABLE_SEARCH_HINT,
+  ASSC_OBJ_TABLE_SUBJ,
+} from '../utils/associated_objects_tab_utils';
+import { DatasourceDetails } from '../../data_connection';
+
+interface AssociatedObjectsTableProps {
+  datasource: DatasourceDetails;
+  selectedDatabase: string;
+  associatedObjects: AssociatedObject[];
+}
+
+interface FilterOption {
+  value: string;
+  text: string;
+}
+
+interface AssociatedTableFilter {
+  type: string;
+  field: string;
+  operator: string;
+  value: string;
+}
+
+export const AssociatedObjectsTable = (props: AssociatedObjectsTableProps) => {
+  const { associatedObjects } = props;
+  const [accelerationFilterOptions, setAccelerationFilterOptions] = useState<FilterOption[]>([]);
+  const [filteredObjects, setFilteredObjects] = useState<AssociatedObject[]>([]);
+
+  const columns = [
+    {
+      field: 'name',
+      name: i18n.translate('datasources.associatedObjectsTab.column.name', {
+        defaultMessage: 'Name',
+      }),
+      sortable: true,
+      'data-test-subj': 'nameCell',
+      render: (name: string, item: AssociatedObject) => (
+        <EuiLink
+          onClick={() => {
+            console.log('name', name);
+            console.log('item', item);
+            if (item.type === 'table') {
+              renderAssociatedObjectsDetailsFlyout(item);
+            } else {
+              renderAccelerationDetailsFlyout(item);
+            }
+          }}
+        >
+          {name}
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'database',
+      name: i18n.translate('datasources.associatedObjectsTab.column.database', {
+        defaultMessage: 'Database',
+      }),
+      sortable: true,
+    },
+    {
+      field: 'type',
+      name: i18n.translate('datasources.associatedObjectsTab.column.type', {
+        defaultMessage: 'Type',
+      }),
+      sortable: true,
+    },
+    {
+      field: 'createdByIntegration',
+      name: i18n.translate('datasources.associatedObjectsTab.column.createdByIntegration', {
+        defaultMessage: 'Created by Integration',
+      }),
+      sortable: true,
+      render: (createdByIntegration: string, _item: AssociatedObject) =>
+        createdByIntegration ? (
+          <EuiLink onClick={() => openDetailsPage(createdByIntegration)}>
+            {createdByIntegration}
+          </EuiLink>
+        ) : (
+          '-'
```

Comment:
nit. thinking should we make this as a const as this seems like a default dom for all other cells.